tsfc/driver.py

```diff
@@ -89,21 +78,24 @@ def compile_integral(integral_data, form_data, prefix, parameters):
 
     # Dict mapping domains to index in original_form.ufl_domains()
     domain_numbering = form_data.original_form.domain_numbering()
-    builder = KernelBuilder(integral_type, integral_data.subdomain_id,
+    builder = KernelBuilder(backend, integral_type, integral_data.subdomain_id,
                             domain_numbering[integral_data.domain])
     return_variables = builder.set_arguments(arguments, argument_indices)
 
     coordinates = ufl_utils.coordinate_coefficient(mesh)
     if ufl_utils.is_element_affine(mesh.ufl_coordinate_element()):
         # For affine mesh geometries we prefer code generation that
         # composes well with optimisations.
-        builder.set_coordinates(coordinates, "coords", mode='list_tensor')
+        builder.set_coordinates(coordinates, mode='list_tensor')
     else:
         # Otherwise we use the approach that might be faster (?)
-        builder.set_coordinates(coordinates, "coords")
+        builder.set_coordinates(coordinates)
 
     builder.set_coefficients(integral_data, form_data)
 
+    builder.set_cell_orientations()
+    builder.set_facets()
```

Comment:
I think these could be just set up in the constructor of the builder without explicitly asking.